src/config.rs

```diff
@@ -881,6 +882,22 @@ impl Config {
         Ok(())
     }
 
+    pub fn cluster(
```

Comment:
A bit unsure about method name (e.g. could be `with_cluster` or `add_cluster`), and if we should "pollute" `Config::` with builder methods or if it's worth having a `ConfigBuilder` with a `build() -> Config`?

Reply:
I think the method name is fine and having the builder methods direct is fine. Having `ConfigBuilder` would make sense if we need to do some initialisation logic (e.g. if the `build` method can fail because it sets stuff up), but I don't think we need that for `Config` since it is just a data store.